Image.php

```diff
@@ -104,11 +104,36 @@ class Image
     protected $cache;
```

Comment:
Looks like the cache attribute is not defaulted to null
It would be a better practice since you test that on L113